apps/website/src/components/shared/form/MapPickerField.tsx

```diff
@@ -0,0 +1,299 @@
+          if (isDraggingRef.current) return;
+          handleLocationSet(map, lat, lng);
+        })
+        // To avoid setting post location on mouse Dragging.
+        .on("dragstart", () => (isDraggingRef.current = true))
+        .on("dragend", () => (isDraggingRef.current = false))
+        .on("zoom", () => {
+          // I don't like this solution, but the ScrollZoomHandler doesn't have this functionality.
+          if (map.getZoom() > maxZoom!) {
+            map.setZoom(maxZoom!);
+          }
+        });
+      return map;
+    }
+  }, [antialias, center, initialZoom, minZoom, maxZoom]);
+
+  /**
+   * Clears the markers and resets the location
+   */
+  const handleLocationClear = useCallback(() => {
+    setPostLocation({} as MapLocation);
+    markersRef.current.forEach((m) => m.remove());
+    markersRef.current = [];
+  }, []);
+
+  /**
+   * Main method that handles marker position and coordinate rounding
+   * @param coords Coords where we want to put the marker
+   * @param map The rendered map
+   */
+  const handleLocationSet = useCallback(
+    async (map: Map, lat: number, lon: number, location?: string) => {
+      console.log("HANDLING LOCATION SET");
+      const roundedCoords = {
+        lat: roundCoord(lat, coordsPrecission),
+        lon: roundCoord(lon, coordsPrecission),
+      };
+      if (!location) {
+        console.log("SEARCHING PLACE NAME...");
+        location = await reverseSearch(roundedCoords.lat, roundedCoords.lon);
+        console.log(`GOT ${location}`);
+      }
+
+      if (!allowMultipleMarkers && markersRef.current.length > 0) {
+        // There's already a marker on the map and we don't allow multiple, so we just update its location.
+        markersRef.current[0]?.setLngLat(roundedCoords);
+      } else {
+        // There are no markers yet or we allow multiple, so let's create one.
+        const marker = getDefaultMarker(roundedCoords, mapRef.current);
+
+        if (marker) {
+          marker
+            .on("dragstart", () => (isDraggingRef.current = true))
+            .on("dragend", () => {
+              console.log("DRAG ENDED ON MARKER 2");
+              handleLocationSet(
+                map,
+                marker.getLngLat().lat,
+                marker.getLngLat().lng,
+              );
+              isDraggingRef.current = false;
+            });
+          markersRef.current.push(marker);
+        }
+      }
+
+      const newPostLocation = {
+        latitude: roundedCoords.lat,
+        longitude: roundedCoords.lon,
+        location: location,
+      };
+
+      setPostLocation(newPostLocation);
+      onLocationChange(newPostLocation);
+    },
+    [coordsPrecission, allowMultipleMarkers, onLocationChange],
+  );
+
+  useEffect(() => {
+    // Ensure map container exists and map hasn't been initialized yet
+    if (!showMap || !mapContainerRef.current || mapRef.current) {
+      return;
+    }
+
+    const map = createMap();
```

Comment:
Inline the creation here.

```suggestion
    const map = new Map({
```